internal/mdxc/application_permission.go

```diff
@@ -163,43 +164,48 @@ func runApplicationPermissionFunctionAzure(function applicationPermissionFunctio
 }
 
 // // -------------- GCP --------------
-// type applicationPermissionFunctionGCP func(context.Context, *gcp.ApplicationPermissionConfig, *iam.Service) error
-
-// func convertApplicationPermissionConfigTerraformToGCP(d *ApplicationPermissionData, a *gcp.ApplicationPermissionConfig, c *gcp.GCPConfig) {
-// 	a.ID = d.Id.Value
-// 	if d.GCPInput != nil {
-// 		a.RoleName = d.GCPInput.RoleName.Value
-// 		a.Condition = d.GCPInput.Condition.Value
-// 	}
-// }
-
-// func convertApplicationPermissionConfigGCPToTerraform(a *gcp.ApplicationPermissionConfig, d *ApplicationPermissionData) {
-// 	d.Id = types.String{Value: a.ID}
-// 	if d.GCPInput == nil {
-// 		d.GCPInput = &GCPApplicationPermissionInputData{}
-// 	}
-// 	d.AzureInput.RoleName = types.String{Value: a.RoleName}
-// 	d.AzureInput.Condition = types.String{Value: a.Condition}
-// }
-
-// func runApplicationPermissionFunctionGCP(function applicationPermissionFunctionGCP, ctx context.Context, d *ApplicationPermissionData, config *gcp.GCPConfig) diag.Diagnostics {
-// 	var diags diag.Diagnostics
-// 	iamClient, serviceErr := config.NewIAMService(ctx)
-// 	if serviceErr != nil {
-// 		diags.Append(
-// 			diag.NewErrorDiagnostic(serviceErr.Error(), ""),
-// 		)
-// 		return diags
-// 	}
-// 	cloudApplicationPermissionConfig := gcp.ApplicationPermissionConfig{}
-// 	convertApplicationPermissionConfigTerraformToGCP(d, &cloudApplicationPermissionConfig, config)
-// 	err := function(ctx, &cloudApplicationPermissionConfig, iamClient)
-// 	if err != nil {
-// 		diags.Append(
-// 			diag.NewErrorDiagnostic(err.Error(), ""),
-// 		)
-// 		return diags
-// 	}
-// 	convertApplicationPermissionConfigGCPToTerraform(&cloudApplicationPermissionConfig, d)
-// 	return diags
-// }
+type applicationPermissionFunctionGCP func(context.Context, *gcp.ApplicationPermissionConfig, gcp.GCPResourceManagerIface) error
+
+func convertApplicationPermissionConfigTerraformToGCP(d *ApplicationPermissionData, a *gcp.ApplicationPermissionConfig, c *gcp.GCPConfig) {
+	a.ID = d.Id.Value
+	a.Project = c.Provider.Project.Value
+	if d.GCPInput != nil {
+		a.Role = d.GCPInput.Role.Value
+		a.Member = d.ApplicationIdentityID.Value
```

Comment:
Is this the "ServiceAccountID" from earlier? If so, let's move the ID field into the nested GCP output for `application_identity` and reference it that way.